### PR TITLE
Feature/homework4

### PR DESCRIPTION
**Pull request by mariami-tchanturia**

### CODE REVIEW

**Review by stepan-lagunovsky — CHANGES_REQUESTED** [review]

**Review by stepan-lagunovsky — COMMENTED** [review]
Believe that now - it's time for "husky" and "lint-staged".

I didn't leave a few comments on purpose so you can set up these two packages correctly and try to detect these errors on commit.
(A tiny note here. Please move settings for lint-staged into .lintstagedrc)

BTW. If we make some lint improvements - lets do the folowing:
1) Move your .eslintrc.json to the root and move "eslintConfig" piece from "package.json" there.
2) For browser list you can install 'browserslist' from npm and move browserlist settings to '.browserlistrc'. Just check the docks.

.browserlistrc ->

```
[production]> 1%
>0.2%
not dead
not op_mini all

[development]
last 1 chrome version
last 1 firefox version
last 1 safari version
```


and etc. For now just this.

An idea is to keep package.json only for package managing stuff.

File: src/App.js [hunk 1]
```diff
@@ -2,29 +2,34 @@ import './App.scss';
 import { useState } from 'react';
 
 import { Counter } from './components/Counter/Counter';
-import { MoviesListing } from './components/MoviesListing/MoviesListing';
-import { Header } from './components/Header/Header';
-import { Footer } from './components/Footer/Footer';
 
-import { GENRES_OPTIONS, SELECT_OPTIONS } from './constants';
+import MovieListPage from './pages/MovieListPage/MovieListPage';
```
Comment (stepan-lagunovsky) [inline src/App.js:6]: `import { Counter } from './components';`

File: src/common/Dialog/Dialog.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { createPortal } from 'react-dom';
```
Comment (stepan-lagunovsky) [inline src/common/Dialog/Dialog.jsx:1]:
```
import { createPortal } from 'react-dom';
import FocusTrap from 'focus-trap-react';
import PropTypes from 'prop-types';

import { Button } from '../../common';
import styles from './Dialog.module.scss';
import closeIcon from '../../assets/close-icon.png';
```

File: src/components/MovieForm/MovieForm.stories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { MovieForm } from '../../components/MovieForm/MovieForm';
```
Comment (stepan-lagunovsky) [inline src/components/MovieForm/MovieForm.stories.js:1]: import { MovieForm } from '../../components';

File: src/components/MovieTile/MovieTile.jsx [hunk 4]
```diff
@@ -1,18 +1,34 @@
+import { useState } from 'react';
 import PropTypes from 'prop-types';
+
 import { MovieMenu } from '../../components/MovieMenu/MovieMenu';
+import { DeleteConfirmation } from '../../components/DeleteConfirmation/DeleteConfirmation';
```
Comment (stepan-lagunovsky) [inline src/components/MovieTile/MovieTile.jsx:5]: import { DeleteConfirmation, MovieMenu } from '../../components';

File: src/pages/MovieListPage/MovieListPage.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { MoviesListing } from '../../components/MoviesListing/MoviesListing';
+import { Header } from '../../components/Header/Header';
+import { Footer } from '../../components/Footer/Footer';
```
Comment (stepan-lagunovsky) [inline src/pages/MovieListPage/MovieListPage.jsx:3]: import { Header, Footer, MoviesListing } from '../../components';

**Review by stepan-lagunovsky — CHANGES_REQUESTED** [review]

File: src/common/Textarea/Textarea.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        required={required}
+        value={value}
+        onChange={onChange && ((e) => onChange(e.target.value))}
+      />
+    </div>
+  );
+};
+
+Textarea.propTypes = {
+  labelText: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (mariami-tchanturia) [reply src/common/Textarea/Textarea.jsx:33]: LabelText in Textarea component isn't required.

File: src/components/Header/Header.jsx [hunk 7]
```diff
@@ -1,32 +1,36 @@
+import { useState } from 'react';
 import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
```
Comment (stepan-lagunovsky) [inline src/components/Header/Header.jsx:3]: 'PropTypes' is defined but never used.

File: src/components/Header/Header.jsx [hunk 8]
```diff
@@ -1,32 +1,36 @@
...
 
-    <div className={`container-lg`}>
-      {preview ? (
-        <MovieDetail movie={preview} />
-      ) : (
+        <Button className='btn--secondary' onClick={() => toggleAddMovie(true)}>
+          + Add Movie
+        </Button>
+      </div>
+      <div className={`container-lg`}>
```
Comment (stepan-lagunovsky) [inline src/components/Header/Header.jsx:33]: 1) You can update this into className="container-lg".

2)There is no such `className` on the next line:
`styles.titleSearchWrapper`
Just `className='container-md'`

File: src/components/MovieForm/MovieForm.stories.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { MovieForm } from '../../components';
```
Comment (stepan-lagunovsky) [inline src/components/MovieForm/MovieForm.stories.js:1]: Just like this:
`import { MovieForm } from './MovieForm';`

File: src/components/MovieMenu/MovieMenu.jsx [hunk 10]
```diff
@@ -2,7 +2,6 @@ import { useState } from 'react';
 import PropTypes from 'prop-types';
 
 import { Button } from '../../common/Button/Button';
```
Comment (stepan-lagunovsky) [inline src/components/MovieMenu/MovieMenu.jsx:4]: import { Button } from '../../common';

**Review by stepan-lagunovsky — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.jsx (hunk 8 above)
Comment (mariami-tchanturia) [reply src/components/Header/Header.jsx]: Fixed

File: src/components/MovieMenu/MovieMenu.jsx (hunk 10 above)
Comment (mariami-tchanturia) [reply src/components/MovieMenu/MovieMenu.jsx]: Fixed

File: src/components/MovieForm/MovieForm.stories.js (hunk 9 above)
Comment (mariami-tchanturia) [reply src/components/MovieForm/MovieForm.stories.js]: Fixed

File: src/common/Textarea/Textarea.jsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  placeholderText,
+  minLength,
+  className = '',
+  required,
+  name,
+  value,
+  onChange,
+}) => {
+  return (
+    <div className={styles.textareaWrapper}>
```
[the remark this answers is not in the document]
Comment (mariami-tchanturia) [reply src/common/Textarea/Textarea.jsx]: Fixed

File: src/components/Header/Header.jsx (hunk 7 above)
Comment (mariami-tchanturia) [reply src/components/Header/Header.jsx:2]: Fixed

**Review by stepan-lagunovsky — APPROVED** [review]

File: src/components/MovieForm/MovieForm.jsx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+          value={plot}
+          onChange={(value) => handleChange('plot', value)}
+        />
+      </div>
+
+      <div className={styles.formActions}>
+        <Button className='btn--default' onClick={handleReset}>
+          Reset
+        </Button>
+        <Button className='btn--primary' onClick={() => handleSubmit(formData)}>
```
Comment (stepan-lagunovsky) [inline src/components/MovieForm/MovieForm.jsx:96]: Please remove onClick

**Review by stepan-lagunovsky — CHANGES_REQUESTED** [review]

**Review by stepan-lagunovsky — CHANGES_REQUESTED** [review]

**Review by stepan-lagunovsky — APPROVED** [review]

**Pull request merged**